rust_icu_ecma402/src/listformat.rs

```diff
@@ -0,0 +1,234 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Implements the traits found in [ecma402_traits::listformat].
+
+use ecma402_traits;
+use ecma402_traits::listformat;
+use rust_icu_common as common;
+use rust_icu_ulistformatter as ulfmt;
+use std::fmt;
+
+/// Implements [listformat::Format] using ICU as the underlying implementing library.
+#[derive(Debug)]
+pub struct Format {
+    rep: ulfmt::UListFormatter,
+}
+
+// Full support for styled formatting is available since v67.
+#[cfg(feature = "icu_version_67_plus")]
+pub(crate) mod internal {
+    use rust_icu_sys as usys;
+    use ecma402_traits::listformat::options;
+
+    // Converts the param types from ECMA-402 into ICU.
+    pub fn to_icu_width(style: &options::Style) -> usys::UListFormatterWidth {
+        use options::Style;
+        use usys::UListFormatterWidth;
+        match style {
+            Style::Long => UListFormatterWidth::ULISTFMT_WIDTH_WIDE,
+            Style::Short => UListFormatterWidth::ULISTFMT_WIDTH_SHORT,
+            Style::Narrow => UListFormatterWidth::ULISTFMT_WIDTH_NARROW,
+        }
+    }
+
+    // Converts the Type param from ECMA-402 into ICU.
+    pub fn to_icu_type(in_type: &options::Type) -> usys::UListFormatterType {
+        use options::Type;
+        use usys::UListFormatterType;
+        match in_type {
+            Type::Conjunction => UListFormatterType::ULISTFMT_TYPE_AND,
+            Type::Disjunction => UListFormatterType::ULISTFMT_TYPE_OR,
+        }
+    }
+}
+
+impl listformat::Format for Format {
```

Comment:
`impl listformat::ListFormatType for ListFormat`.

Reply:
Since the conclusions will likely be the same for this comment as the one above, I'll resolve this conversation so we don't have two threads about what amounts to the same thing.